execution_engine/src/runtime/mod.rs

```diff
@@ -3223,4 +3242,72 @@ where
             None => Err(Error::InvalidContract(contract_hash)),
         }
     }
+
+    fn add_message_topic(&mut self, topic_name: String) -> Result<Result<(), ApiError>, Error> {
+        let topic_hash = crypto::blake2b(AsRef::<[u8]>::as_ref(&topic_name)).into();
```

Comment:
```suggestion
        let topic_hash = crypto::blake2b(topic_name.as_bytes()).into();
```

Reply:
actually it seems that just borrowing should work 🤔 because `blake2b` is defined as `blake2b<T: AsRef<[u8]>>(data: T)`
```suggestion
        let topic_hash = crypto::blake2b(&topic_name).into();
```

Reply:
Changed here: https://github.com/casper-network/casper-node/pull/4315/commits/566b3534a29355dd56ec8375f4c42d0a2ca0f24c